packages/engine/Source/Renderer/ShaderSource.js

```diff
@@ -278,19 +266,38 @@ function combineShader(shaderSource, isFragmentShader, context) {
   }
 
   // append built-ins
+  let builtinSources = "";
   if (shaderSource.includeBuiltIns) {
-    result += getBuiltinsAndAutomaticUniforms(combinedSources);
+    builtinSources = getBuiltinsAndAutomaticUniforms(combinedSources);
   }
 
   // reset line number
   result += "\n#line 0\n";
 
   // append actual source
+  const combinedShader = builtinSources + combinedSources;
+  if (
+    context.webgl2 &&
+    isFragmentShader &&
+    !/layout\s*\(location\s*=\s*0\)\s*out\s+vec4\s+out_FragColor;/g.test(
+      combinedShader
+    ) &&
+    !/czm_out_FragColor/g.test(combinedShader) &&
+    /out_FragColor/g.test(combinedShader)
+  ) {
+    result += "layout(location = 0) out vec4 out_FragColor;\n\n";
+  }
```

Comment:
Should the shader be responsible for adding `layout(location = 0) out vec4 out_FragColor;`?